ddp-workspace/projects/ddp-sdk/src/lib/services/cookiesManagement.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Inject, Injectable } from '@angular/core';
+import { CookieService } from 'ngx-cookie';
+import { BehaviorSubject } from 'rxjs';
+import { ConfigurationService } from './configuration.service';
+import { AnalyticsEventsService } from './analyticsEvents.service';
+
+@Injectable()
+export class CookiesManagementService {
+  private isFirstTimeVisitor: BehaviorSubject<boolean> =  new BehaviorSubject(null);
+  private isCookiesToReject: BehaviorSubject<boolean> =  new BehaviorSubject(null);
+  private expireDays = 10;
+
+  constructor(private cookie: CookieService,
+              private analytics: AnalyticsEventsService,
+              @Inject('ddp.config') private configuration: ConfigurationService) {
+  }
+
+  public manageCookies(): void {
+    this.checkCookiesStatus();
+    this.checkCookiesToReject();
+    this.checkFirstTimeCookie();
+  }
+
+  private checkCookiesStatus(): void {
+    const status = this.cookie.get('consent');
+    if (status === null || status === undefined) {
+      this.cookie.put('consent', 'false', {expires: this.getExpirationDate()});
+    }
+  }
+
+  private checkCookiesToReject(): void {
+    // If there are cookies other then functional -> they can be rejected -> show banner
+    this.configuration.cookies.cookies.filter(x => x.type !== 'Functional').length
+      ? this.isCookiesToReject.next(true)
+      : this.isCookiesToReject.next(false);
+  }
+
+  private checkFirstTimeCookie(): void {
+    if (this.cookie.get('isFirstTimeVisitor') === null || this.cookie.get('isFirstTimeVisitor') === undefined) {
+      this.cookie.put('isFirstTimeVisitor', 'true', {expires: this.getExpirationDate()});
+      this.isFirstTimeVisitor.next(true);
+    } else {
+      this.cookie.get('isFirstTimeVisitor') === 'true'
+        ? this.isFirstTimeVisitor.next(true)
+        : this.isFirstTimeVisitor.next(false);
+    }
+  }
+
+  public getIsCookiesToReject(): BehaviorSubject<boolean> {
+    return this.isCookiesToReject;
+  }
+
+  public getIsFirstTimeVisitor(): BehaviorSubject<boolean> {
+    return this.isFirstTimeVisitor;
+  }
+
+  public closeBanner(): void {
+    this.cookie.put('isFirstTimeVisitor', 'false', {expires: this.getExpirationDate()});
+    this.isFirstTimeVisitor.next(false);
+    this.isCookiesToReject.next(false);
+  }
+
+  public acceptAll(): void {
```

Comment:
Note this in AC5: "If no changes are made via Manage Preferences, place all available cookies."  Presumably a specific cookie could be rejected in Manage Preferences and then the user could click accept, in which case we would want to accept only the cookies not specifically rejected.  You wouldn't necessarily need to implement that full thing in this ticket, but I think this might change how you'd write this particular method.

Reply:
--in progress--